### PR TITLE
destination: Split GetProfile into smaller functions

### PR DESCRIPTION
**Pull request by olix0r**
Before changing any GetProfile behavior, this change splits the API handler into some smaller scopes. This helps to clarify control flow and reduce nested contexts. This change also adds relevant fields to log contexts to improve diagnostics.

Via https://github.com/linkerd/linkerd2/pull/10498

### CODE REVIEW

**Review by adleong — APPROVED** [review]
nice cleanup

File: controller/api/destination/server.go [hunk 1]
```diff
@@ -338,41 +299,82 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
...
 	select {
 	case <-s.shutdown:
 	case <-stream.Context().Done():
-		log.Debugf("GetProfile(%+v) cancelled", dest)
+		log.Debug("Cancelled")
 	}
+	return nil
+}
 
+func (s *server) translateEndpointProfile(
```
[the remark this answers is not in the document]
Comment (olix0r) [reply controller/api/destination/server.go:337]: cool i'll pull in some commentary/cleanups into the followup(s).

File: controller/api/destination/server.go [hunk 2]
```diff
@@ -338,41 +299,82 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
...
+	opaquePorts, err := getAnnotatedOpaquePorts(address.Pod, s.defaultOpaquePorts)
+	if err != nil {
+		return fmt.Errorf("failed to get opaque ports for pod: %w", err)
+	}
+	var endpoint *pb.WeightedAddr
+	endpoint, err = s.createEndpoint(*address, opaquePorts)
+	if err != nil {
+		return fmt.Errorf("failed to create endpoint: %w", err)
+	}
+	translator := newEndpointProfileTranslator(address.Pod, port, endpoint, stream, s.log)
```
Comment (alpeb) [inline controller/api/destination/server.go:358]:
```suggestion
	translator := newEndpointProfileTranslator(address.Pod, port, endpoint, stream, log)
```

Comment (olix0r) [reply controller/api/destination/server.go:358]: thanks i'll include that in the follow up

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**